x-pack/plugins/rule_registry/common/assets/field_maps/technical_rule_field_map.test.ts

```diff
@@ -37,6 +37,11 @@ it('matches snapshot', () => {
         "required": false,
         "type": "keyword",
       },
+      "kibana.alert.cases.id": Object {
```

Comment:
I am am wondering if we should use `nested field` for case and having something like that
```
"kibana.alert.cases": {
          "type": "nested", 
          "properties": {
            "name":    { "type": "keyword"  },
            "id": { "type": "keyword"  },
          }
        }
```
and maybe we should add `added_by` and `added_at`

Reply:
I think we can do it like `kibana.alert.cases.added_by: { ... }` and `kibana.alert.cases.added_at: { ... }`. The `nested` field is only necessary if we need to query cases inside the array independently from each other which I don't think we need it.

Reply:
Agree, nested is only necessary if we need to query by both ID & title. Is it possible a user could add a KQL filter for `kibana.alert.cases.id: "abc" AND kibana.alert.cases.name: "test"` and be confused by it returns multiple cases with the name `test`?

Reply:
We will not add the tile of the case inside the schema for privacy reasons (a user with read access to alerts and no access to cases should not see the case title). Only the id will be saved in the alerts schema.

Reply:
I think `kibana.alert.cases.id: "abc" AND kibana.alert.cases.name: "test"` will return only one case because the id is unique, correct?

Reply:
Though, the search is based on alerts so it will return all alerts that are attached to the case with id "abc" and title "test"

Reply:
If you are storing the info in an array like:

```
[
  { "id": "abc", name: "test" },
  { "id": "def", name: "test" }
]
```
and this is NOT indexed as nested, I believe Elasticsearch does not behave as expected. See https://stackoverflow.com/questions/65152437/elasticsearch-multi-field-search-in-non-nested-arrays.

Reply:
> We will not add the tile of the case inside the schema for privacy reasons (a user with read access to alerts and no access to cases should not see the case title). Only the id will be saved in the alerts schema.

Does that mean you're doing a secondary query to get the title of the case to show in the alert table?

Reply:
Oh yeah, I see what you mean. You are both right. It will return alerts that have a case with the name "test" even though the ID is different. Even if we do not add the title this may occur if we add `added_by` and `added_at`. IMO, it is not worth the complexity added by nested fields to support this use case. What do you think? Also, does KQL support nested queries?

>Does that mean you're doing a secondary query to get the title of the case to show in the alert table?

Yes, we will fetch the title and the owner of the cases shown in the table. We discuss it in this issue https://github.com/elastic/kibana/issues/146864 and we decided that it is ok to do so. What do you think about it?

@XavierM About `added_by` and `added_at`, what value do you have in mind for the `added_by`? What do we do for the `rule.created_by` field?

Reply:
I updated the schema as we discussed offline.